gnomad_qc/v4/create_release/create_release_sites_ht_exomes.py

```diff
@@ -0,0 +1,761 @@
+            calling_intervals(interval_name="ukb", calling_interval_padding=50).ht()[
+                ht.locus
+            ]
+        ),
+        outside_broad_capture_region=~hl.is_defined(
+            calling_intervals(interval_name="broad", calling_interval_padding=50).ht()[
+                ht.locus
+            ]
+        ),
+    )
+
+    return selects
+
+
+def custom_filters_select(ht: hl.Table) -> Dict[str, hl.expr.Expression]:
+    """
+    Select gnomAD filter HT fields for release dataset.
+
+    Extract "results" field and rename based on filtering method.
+
+    :param ht: Filters Hail Table.
+    :return: Select expression dict.
+    """
+    filter_name = hl.eval(ht.filtering_model.filter_name)
+    if filter_name == "RF":
+        name = "random_forest_results"
+    elif filter_name == "AS_VQSR":
+        name = "vqsr_results"
+    elif filter_name == "IF":
+        name = "isolation_forest_results"
+    else:
+        raise ValueError(f"Filtering method {filter_name} not recognized.")
+    selects = {name: ht.results.annotate(**ht.training_info)}
+
+    return selects
+
+
+def custom_info_select(ht: hl.Table) -> Dict[str, hl.expr.Expression]:
+    """
+    Select fields for info Hail Table annotation in release.
+
+    The info field requires fields from the freq HT and the filters HT so those are
+    pulled in here along with all info HT fields. It also adds the `allele_info` struct
+    to release HT.
+
+    :param ht: Info Hail Table.
+    :return: Select expression dict.
+    """
+    # Create a dict of the fields from the filters HT that we want to add to the info.
+    filters_ht = get_config().get("filters")["ht"]
+
+    # For more information on the selected compute_info_method, please see the
+    # run_compute_info in gnomad_qc.v4.annotations.generate_variant_qc_annotations.py
+    compute_info_method = hl.eval(
+        filters_ht.filtering_model_specific_info.compute_info_method
+    )
+    score_name = hl.eval(filters_ht.filtering_model.score_name)
+    filters_ht = filters_ht.transmute(**filters_ht.truth_sets)
+    filters = filters_ht[ht.key]
+    filters_info_fields = [
+        "singleton",
+        "transmitted_singleton",
+        "omni",
+        "mills",
+        "monoallelic",
+        "only_het",
+    ]
+    filters_info_dict = {field: filters[field] for field in filters_info_fields}
+    filters_info_dict.update({**{f"{score_name}": filters[f"{score_name}"]}})
+
+    # Create a dict of the fields from the freq HT that we want to add to the info.
+    freq_ht = get_config().get("freq")["ht"]
+    freq_info_dict = {"inbreeding_coeff": freq_ht[ht.key]["inbreeding_coeff"]}
+
+    # Create a dict of the fields from the VRS HT that we want to add to the info.
+    vrs_ht = get_vrs().ht()
+    vrs_info_fields = {"vrs": vrs_ht[ht.key].vrs}
+
+    # Create a dict of the fields from the info HT that we want keep in the info.
+    info_struct = hl.struct(**ht.site_info, **ht[f"{compute_info_method}_info"])
```

Comment:
Here `ht` still means `get_info().ht()`, right? Does it have `site_info`?

Reply:
ht here is `hl.read_table(ht_path, _intervals=_intervals)` from the get_ht function. ht_path would be `get_info().path`. It's basically equivalent to get_info().ht() but its read in with the same intervals as freq to improve the join efficiency.  Yes it does have a `site_info`.